util-images/network/netperfbenchmark/pkg/worker/util.go

```diff
@@ -0,0 +1,200 @@
+)
+
+// Function to be applied for each metric for aggregation
+var (
+	iperfUDPFunction = []string{"", "", "", "Sum", "Sum", "Sum", "Sum", "Sum", "Avg", "Avg", "Min", "Max", "Avg", "Sum"}
+	iperfTCPFunction = []string{"", "", "", "Sum", "Avg"}
+)
+
+// parseResult parses the response received for each protocol type
+func parseResult(protocol string, result []string, testDuration string) ([]float64, error) {
+	switch protocol {
+	case ProtocolTCP:
+		return parseIperfResponse(result, testDuration, ProtocolTCP, tcpMetricsCount, zeroFormatTCP, fractionZeroFormatTCP, iperfTCPFunction)
+	case ProtocolUDP:
+		return parseIperfResponse(result, testDuration, ProtocolUDP, udpMetricsCount, zeroFormatUDP, fractionZeroFormatUDP, iperfUDPFunction)
+	case ProtocolHTTP:
+		return parseSiegeResponse(result)
+	default:
+		return nil, errors.New("invalid protocol: " + protocol)
+	}
+}
+
+func parseIperfResponse(result []string, testDuration string, protocol string, metricCount int, zeroFormat string, fractionZeroFormat string, operators []string) ([]float64, error) {
+	aggregatedResult := make([]float64, 0, metricCount)
+	count := 0
+	sessionID := make(map[string]bool)
+	interval := zeroFormat + "-" + testDuration + fractionZeroFormat
+	for _, line := range result {
+		split := parseIperfLine(line, zeroFormat, fractionZeroFormat, testDuration)
+		// iperf gives aggregated result in a row with "SUM", but many fields are not getting aggregated
+		// hence not depending on iperf's aggregation
+		if len(split) < metricCount+1 || "SUM" == split[1] || split[2] != interval {
+			continue
+		}
+		// iperf sometimes prints duplicate rows for same session id(for same duration), making sure
+		// the row is taken for calculation only once.
+		if _, ok := sessionID[split[1]]; ok {
+			continue
+		}
+		sessionID[split[1]] = true
+		// The first three and the last items are not metrics in iperf summary.
+		for i := 3; i < len(split)-1; i++ {
+			tmp, err := strconv.ParseFloat(split[i], 64)
+			if err != nil {
+				klog.Errorf("Error converting %v to float: %v", split[i], err)
+				return nil, err
+			}
+			if len(aggregatedResult) < metricCount {
+				aggregatedResult = append(aggregatedResult, tmp)
+			} else {
+				switch operators[i] {
+				case "Sum":
+					aggregatedResult[i-3] = tmp + aggregatedResult[i-3]
+				case "Avg":
+					aggregatedResult[i-3] = (float64(count)*tmp + aggregatedResult[i-3]) / (float64(1 + count))
+				case "Min":
+					aggregatedResult[i-3] = math.Min(tmp, aggregatedResult[i-3])
+				case "Max":
+					aggregatedResult[i-3] = math.Max(tmp, aggregatedResult[i-3])
+				}
+			}
+			count++
+		}
+	}
+	klog.Infof("Final output: %v", aggregatedResult)
+	return aggregatedResult, nil
+}
+
+func parseIperfLine(line string, zeroFormat string, fractionZeroFormat string, testDuration string) []string {
+	formattedString := hyphenSpaceRegex.ReplaceAllString(multiSpaceRegex.ReplaceAllString(unitRegex.ReplaceAllString(line, " "), " "), "-")
```

Comment:
nit: can you please shorten the line by splitting it?
e.g.
```
noUnitsLine := unitRegex.ReplaceAllString(line, " ")
formatterLine := hyphenSpaceRegex.ReplaceAllString(multiSpaceRegex.ReplaceAllString(noUnitsLine, " "), "-")
```

Reply:
Modified.